content/token-swap.md

```diff
@@ -0,0 +1,848 @@
+    100e9,
+)
+
+transaction.add(instruction)
+```
+
+## Curves
+
+Trading curves are at the core of how swap pools and AMMs (Automated Market Makers) operate. The trading curve is the function that the token swap program uses to calculate how much of a destination token will be provided given an amount of source token. The curve effectively sets the market price of the tokens in the pool.
+
+The pool we’ll be interacting with in this lesson employs a [Constant Product](https://spl.solana.com/token-swap#curves) Curve Function. The constant product curve is the well-known Uniswap and Balancer style curve that preserves an invariant on all swaps. This invariant can be expressed as the product of the quantity of token A and token B in the swap pool.
+
+```tsx
+A_total * B_total = invariant
+```
+
+If we have 100 token A and 5,000 token B, our invariant is 500,000.
+
+Now, if a trader wishes to put in a specific amount token A for some amount of token B, the calculation becomes a matter of resolving "B_out" where:
+
+```tsx
+(A_total + A_in) * (B_total - B_out) = invariant
+```
+
+Putting in the 10 token A along with our invariant of half a million, we would need to solve for "B_out" like so:
+
+```tsx
+(100 + 10) * (5,000 - B_out) = 500,000
+5,000 - B_out = 500,000 / 110
+5,000 - (500,000 / 110) = B_out
+B_out = 454.5454...
+```
+
+The product of the amount of token A and token B must always equal a constant, hence the name ‘Constant Product’. More information can be found on the [Uniswap whitepaper](https://uniswap.org/whitepaper.pdf) and the [Balancer whitepaper](https://balancer.fi/whitepaper.pdf).
+
+If curves don't make a whole lot of sense, don't worry! While learning more about how they work doesn't hurt, you don't need to understand the entirety of the mathematics to be able to implement the common curves.
+
+# Demo
+
+For this demo, a token pool of two brand new tokens has been created and is live on Devnet. We'll walk through building out a frontend UI to interact with this swap pool! Since the pool is already made, we don't have to worry about initiating the pool and funding it with tokens. Instead, we'll focus on building out the instructions for
+
+- depositing liquidity to the pool
+- withdrawing your deposited liquidity
+- swapping from one token to the other
+
+![Screenshot of Token Swap Demo](../assets/token-swap-frontend.png)
+
+### 1. Download the starter code
+
+Before we get started, go ahead and download the [starter code](https://github.com/Unboxed-Software/solana-token-swap-frontend/tree/starter).
+
+The project is a fairly simple Next.js application re-using a lot of what was previously built out for the demos in the first lesson. As you can see from the image above, there are a few different text inputs and buttons - all of which will submit transactions to the blockchain on the user's behalf. Our focus in this demo will be creating the instructions that the last three buttons will submit.
```

Comment:
@jamesrp13 re: 1st sentence, "...for the demo in the first lesson."

First "module." ...or maybe, "the first lessons." ?